pennylane/devices/qutrit_mixed/initialize_state.py

```diff
@@ -0,0 +1,119 @@
+# limitations under the License.
+"""Functions to prepare a qutrit mixed state."""
+
+from typing import Iterable, Union
+import numpy as np
+import pennylane as qml
+from pennylane import (
+    QutritBasisState,
+)
+from pennylane.operation import (
+    StatePrepBase,
+    Operation,
+)
+from pennylane.wires import Wires
+
+qudit_dim = 3  # specifies qudit dimension
+
+
+def create_initial_state(
+    wires: Union[qml.wires.Wires, Iterable],
+    prep_operation: Operation = None,
+    like: str = None,
+):
+    r"""
+    Returns an initial state, defaulting to :math:`\ket{0}\bra{0}` if no state-prep operator is provided.
+
+    Args:
+        wires (Union[Wires, Iterable]): The wires to be present in the initial state
+        prep_operation (Optional[StatePrepBase]): An operation to prepare the initial state
+        like (Optional[str]): The machine learning interface used to create the initial state.
+            Defaults to None
+
+    Returns:
+        array: The initial state of a circuit
+    """
+    if isinstance(wires, Wires):
+        wires = wires.tolist()
+
+    num_wires = len(wires)
+
+    if not prep_operation:
+        rho = _create_basis_state(num_wires, 0)
+
+    elif isinstance(prep_operation, QutritBasisState):
+        rho = _apply_basis_state(prep_operation.parameters[0], wires)
+
+    elif isinstance(prep_operation, StatePrepBase):
+        rho = _apply_state_vector(prep_operation.state_vector(wire_order=list(wires)), num_wires)
+
+    # TODO: add instance for prep_operations as added
+
+    return qml.math.asarray(rho, like=like)
+
+
+def _apply_state_vector(state, num_wires):  # function is easy to abstract for qudit
+    """Initialize the internal state in a specified pure state.
+
+    Args:
+        state (array[complex]): normalized input state of length
+            ``3**len(wires)``
+        num_wires (int): number of wires that get initialized in the state
+
+    Returns:
+        array[complex]: complex array of shape ``[3] * (2 * num_wires)``
+        representing the density matrix of this state.
+    """
+
+    # Initialize the entire wires with the state
+    rho = qml.math.outer(state, qml.math.conj(state))
+    return qml.math.reshape(rho, [3] * 2 * num_wires)
+
+
+def _apply_basis_state(state, wires):  # function is easy to abstract for qudit
+    """Returns initial state for a specified computational basis state.
+
+    Args:
+        state (array[int]): computational basis state of shape ``(wires,)``
+            consisting of 0s, 1s and 2s.
+        wires (Iterable[int]): wires that the provided computational state should be initialized on
+
+    Returns:
+        array[complex]: complex array of shape ``[3] * (2 * num_wires)``
+        representing the density matrix of this basis state.
+    """
+    num_wires = len(wires)
+
+    # get computational basis state number
+    basis_states = qudit_dim ** (num_wires - 1 - np.array(wires))
+    num = int(qml.math.dot(state, basis_states))
```

Comment:
```suggestion
    num = int("".join([str(x) for x in state]), qudit_dim)
```
This would accomplish the same thing?

Reply:
That looks right.